exporter/splunkhecexporter/client.go

```diff
@@ -94,6 +97,70 @@ func (c *client) pushMetricsData(
 	return numDroppedTimeseries, nil
 }
 
+func (c *client) pushTraceData(
+	ctx context.Context,
+	td consumerdata.TraceData,
+) (droppedSpans int, err error) {
+	c.wg.Add(1)
+	defer c.wg.Done()
+
+	splunkEvents, numDroppedSpans := traceDataToSplunk(c.logger, td, c.config)
+	if len(splunkEvents) == 0 {
+		return numDroppedSpans, nil
+	}
+
+	body, compressed, err := encodeBodyEvents(&c.zippers, splunkEvents, c.config.DisableCompression)
+	if err != nil {
+		return len(td.Spans), consumererror.Permanent(err)
+	}
+
+	// TODO the client sends synchronously data as of now. It would make more sense to buffer data coming in, and send batches as supported by Splunk. To batch effectively, a ring buffer approach would work well - based on time and number of messages queued.
```

Comment:
This should not be necessary once the exporter is upgraded to use the new trace data type. The new type is capable of representing batches from multiple resources and coupled with recommended `batch` processor should no longer require any batching here.

Reply:
Removed the TODOs for now.